tests/ui/polymorphization/taint.rs

```diff
@@ -1,6 +1,7 @@
-//@ known-bug: rust-lang/rust#126896
 //@ compile-flags: -Zvalidate-mir -Zinline-mir=yes
 
+// reported as rust-lang/rust#126896
```

Comment:
this doesn't belong in `polymorphization`, please put it into the `type_alias_impl_trait` folder

Reply:
Will do. I based that choice on https://github.com/rust-lang/rust/issues/126896#issuecomment-2186182800 saying that the `-Zpolymorphization` is the loadbearing part of that test.

Reply:
`-Zpolymorphization` isn't enabled in this test 😅 I am confused by @lqd's comment, maybe it's missing a 'not'? 😁

But alright, given that context I am actually happy that you've looked at the original issue to figure out wher to put this test. Thank you!

Reply:
so, is there a more appropriate place for this test to live, or is it good where it is?

Reply:
given that it triggers with `-Zvalidate-mir` it isn't related to polymorphization, just triggered by it. THe underlying bug is a TAIT one

Reply:
allright, moved to `tests/ui/type-alias-impl-trait/`